spec/porky_lib/file_service_spec.rb

```diff
@@ -94,240 +95,384 @@ def write_test_file(data)
+        tempfile = write_test_file(File.read(tempfile_path))
+        tempfile.rewind
 
-  it 'file_contents does not contain metadata field if none provided' do
-    file_data = JSON.parse(ciphertext_data, symbolize_names: true)
-    file_contents = file_service.send(:file_contents, file_data[:key], file_data[:data], file_data[:nonce], {})
-    expect(JSON.parse(file_contents, symbolize_names: true)).not_to have_key(:metadata)
-  end
+        message, = file_service.send(:decrypt_file_contents, tempfile)
+        expect(message).to eq(plaintext_data)
+      end
 
-  it 'file_contents does not contain metadata field if options is nil' do
-    file_data = JSON.parse(ciphertext_data, symbolize_names: true)
-    file_contents = file_service.send(:file_contents, file_data[:key], file_data[:data], file_data[:nonce], nil)
-    expect(JSON.parse(file_contents, symbolize_names: true)).not_to have_key(:metadata)
-  end
+      path = write_test_file(plaintext_data).path
+      file_service.write_file(path, bucket_name, default_key_id)
+    end
+    # rubocop:enable RSpec/MessageSpies
 
-  it 'attempt to write with file nil raises FileServiceError' do
-    expect do
-      file_service.write(nil, bucket_name, default_key_id)
-    end.to raise_exception(PorkyLib::FileService::FileServiceError)
-  end
+    it 'raises FileServiceError when file path does not exist' do
+      expect do
+        file_service.write_file('non_existent_path', bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'attempt to write with bucket name nil raises FileServiceError' do
-    expect do
-      file_service.write(plaintext_data, nil, default_key_id)
-    end.to raise_exception(PorkyLib::FileService::FileServiceError)
-  end
+    it 'raises FileServiceError when file is nil ' do
+      expect do
+        file_service.write_file(nil, bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'attempt to write with key ID nil raises FileServiceError' do
-    expect do
-      file_service.write(plaintext_data, bucket_name, nil)
-    end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    it 'raises FileServiceError when bucket name is nil' do
+      expect do
+        file_service.write_file(write_test_file(plaintext_data), nil, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
+
+    it 'raises FileServiceError when key ID is nil ' do
+      expect do
+        file_service.write_file(write_test_file(plaintext_data), bucket_name, nil)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
   end
 
-  it 'attempt to write to bucket without permission raises FileServiceError' do
-    Aws.config[:s3].delete(:stub_responses)
-    Aws.config[:s3] = {
-      stub_responses: {
-        put_object: 'Forbidden'
+  describe '#write_data' do
```

Comment:
do we have tests for `FileSizeTooLargeError` under this `describe '#write_data'` block?

Reply:
There's a test that checks if it handles big files, but not one that throws that exception, will push new tests